Source/processcontainers/ProcessContainer.h

```diff
@@ -27,71 +27,71 @@ namespace ProcessContainers {
     using IStringIterator = Core::IteratorType<std::vector<string>, const string>;
     using IConstStringIterator = Core::IteratorType<const std::vector<string>, const string, std::vector<string>::const_iterator>;
 
-    struct NetworkInterfaceIterator {
-        NetworkInterfaceIterator();
-        virtual ~NetworkInterfaceIterator() {};
-        
-        // Common iterator functions:
-        // -----------------------------------
-        virtual bool Next();
-        virtual void Reset();
-        virtual bool IsValid() const;
-        virtual uint32_t Count() const;
-
-        // Lifetime management:
-        // ------------------------------------
-        void AddRef();
-        void Release();
-        
-        // Implementation-dependent
-        // -----------------------------------
-
+    class INetworkInterfaceIterator : public Core::IIterator 
```

Comment:
Very good to derive from the IIterator but I think we now lost the refcount interface (although still implemented in NetworkInfoUnimplemented.cpp). I would propose to add IReferenceCounted as well